### ISSUE TITLE
Flame Blade memory leak

### ISSUE THREAD

**Issue by DeliciousTea** [ChaosBuffalo/MKUltra#105]
Seems to be an issue with Flame Blade from the druid class that causes severe issues. My server crashed from this and I tested it in single player and had no specific error but my game immediately started to lag until it froze completely and had to be terminated in task manager.

We are using forge 14.23.5.2836 and the latest MK Ultra release, version 0.81.

The console on my server reads:
```
4:44:13 PM
 CONSOLE:
[WARN] MKAREA instant null cast! Spell: effect.flame_blade_effect
4:44:13 PM
 CONSOLE:
[WARN] MKAREA instant null cast! Spell: effect.particle_potion
4:44:13 PM
 CONSOLE:
[WARN] MKAREA instant null cast! Spell: effect.flame_blade_effect
4:44:13 PM
 CONSOLE:
[WARN] MKAREA instant null cast! Spell: effect.particle_potion
4:44:27 PM
 CONSOLE:
[WARN] Tried to get a spell on an unregistered target! Spell: effect.flame_blade
```

And here's the error from the crash report.
```
java.lang.Error: ServerHangWatchdog detected that a single server tick took 60.00 seconds (should be max 0.05)
    at net.minecraft.entity.Entity.<init>(Entity.java:198)
    at com.chaosbuffalo.mkultra.entities.EntityMKAreaEffect.<init>(EntityMKAreaEffect.java:71)
    at com.chaosbuffalo.mkultra.entities.EntityMKAreaEffect.<init>(EntityMKAreaEffect.java:83)
    at com.chaosbuffalo.mkultra.effects.AreaEffectBuilder.<init>(AreaEffectBuilder.java:17)
    at com.chaosbuffalo.mkultra.effects.AreaEffectBuilder.Create(AreaEffectBuilder.java:22)
    at com.chaosbuffalo.mkultra.effects.spells.FlameBladePotion.onAttackEntity(FlameBladePotion.java:54)
    at com.chaosbuffalo.mkultra.effects.spells.FlameBladePotion$$Lambda$1002/901052907.apply(Unknown Source)
    at com.chaosbuffalo.mkultra.effects.SpellTriggers$ATTACK_ENTITY.lambda$onAttackEntity$0(SpellTriggers.java:250)
    at com.chaosbuffalo.mkultra.effects.SpellTriggers$ATTACK_ENTITY$$Lambda$1991/1022281670.accept(Unknown Source)
    at java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
    at com.chaosbuffalo.mkultra.effects.SpellTriggers$ATTACK_ENTITY.onAttackEntity(SpellTriggers.java:247)
    at com.chaosbuffalo.mkultra.event.CombatEventHandler.onLivingAttackEvent(CombatEventHandler.java:66)
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_383_CombatEventHandler_onLivingAttackEvent_LivingAttackEvent.invoke(.dynamic)
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
    at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
    at net.minecraftforge.common.ForgeHooks.onLivingAttack(ForgeHooks.java:585)
    at net.minecraft.entity.EntityLivingBase.attackEntityFrom(EntityLivingBase.java:857)
    at com.chaosbuffalo.mkultra.effects.spells.FlameBladeEffectPotion.doEffect(FlameBladeEffectPotion.java:48)
    at com.chaosbuffalo.mkultra.entities.EntityMKAreaEffect.applyEffectsToTarget(EntityMKAreaEffect.java:279)
    at com.chaosbuffalo.mkultra.entities.EntityMKAreaEffect.onUpdate(EntityMKAreaEffect.java:222)
    at net.minecraft.world.World.updateEntityWithOptionalForce(World.java:1993)
    at net.minecraft.world.WorldServer.updateEntityWithOptionalForce(WorldServer.java:832)
    at net.minecraft.world.World.updateEntity(World.java:1955)
    at net.minecraft.world.World.redirect$updateEntity$zzj000(World.java:4118)
    at net.minecraft.world.World.updateEntities(World.java:1759)
    at net.minecraft.world.WorldServer.updateEntities(WorldServer.java:613)
    at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:767)
    at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:397)
    at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
    at java.lang.Thread.run(Thread.java:748)
```

**Comment by DeliciousTea** [comment]
I've done some additional testing with this and I'm not sure if it occurs with just MK Ultra installed or not. I've got a fairly large pack so it could just become apparent due to the memory usage of other mods. I've definitely noticed that this issue comes up with different buff spells. There may be some connection to various tweak mods like BetterFPS or Foamfix, but I'm still not 100% sure. I'll keep updating here as I find more info.

**Comment by Kovak** [comment]
This is fixed in the 0.82 release on curseforge.
